pkg/operator/backupcontroller/backupcontroller.go

```diff
@@ -0,0 +1,364 @@
+	if err != nil {
+		return fmt.Errorf("error while updating backup status [%s]: %w", backup.Name, err)
+	}
+
+	if job.Labels == nil {
+		job.Labels = map[string]string{}
+	}
+
+	job.Labels["state"] = "processed"
+	_, err = jobClient.Update(ctx, &job, v1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("error while updating job labels [%s]: %w", job.Name, err)
+	}
+
+	return nil
+}
+
+func createBackupJob(ctx context.Context,
+	backup operatorv1alpha1.EtcdBackup,
+	targetImagePullSpec string,
+	jobClient batchv1client.JobInterface,
+	backupClient operatorv1alpha1client.EtcdBackupInterface) error {
+	scheme := runtime.NewScheme()
+	codec := serializer.NewCodecFactory(scheme)
+	err := batchv1.AddToScheme(scheme)
+	if err != nil {
+		return fmt.Errorf("BackupController could not add batchv1 scheme: %w", err)
+	}
+
+	obj, err := runtime.Decode(codec.UniversalDecoder(batchv1.SchemeGroupVersion), etcd_assets.MustAsset("etcd/cluster-backup-job.yaml"))
+	if err != nil {
+		return fmt.Errorf("BackupController could not decode batchv1 job scheme: %w", err)
+	}
+
+	backupFileName := fmt.Sprintf("backup-%s-%s", backup.Name, time.Now().Format("2006-01-02_150405"))
+
+	job := obj.(*batchv1.Job)
+	job.Name = names.SimpleNameGenerator.GenerateName(job.Name + "-")
+	job.Labels[backupJobLabel] = backup.Name
+	job.OwnerReferences = append(job.OwnerReferences, v1.OwnerReference{
+		APIVersion: backup.APIVersion,
+		Kind:       backup.Kind,
+		Name:       backup.Name,
+		UID:        backup.UID,
+	})
+
+	job.Spec.Template.Spec.Containers[0].Image = targetImagePullSpec
+	job.Spec.Template.Spec.Containers[0].Env = []corev1.EnvVar{
+		{Name: backupDirEnvName, Value: fmt.Sprintf("%s/%s", recentBackupPath, backupFileName)},
+	}
+
+	injected := false
+	for _, mount := range job.Spec.Template.Spec.Volumes {
+		if mount.Name == "etc-kubernetes-cluster-backup" {
+			mount.PersistentVolumeClaim.ClaimName = backup.Spec.PVCName
+			injected = true
+			break
+		}
+	}
+
+	if !injected {
+		return fmt.Errorf("could not inject PVC into Job template, please check the included cluster-backup-job.yaml")
+	}
+
+	klog.Infof("BackupController starts with backup [%s] as job [%s], writing to filename [%s]", backup.Name, job.Name, backupFileName)
+	_, err = jobClient.Create(ctx, job, v1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("BackupController could create job: %w", err)
+	}
+
+	backup.Status = operatorv1alpha1.EtcdBackupStatus{
+		Conditions: []v1.Condition{
+			{
+				Type:               "BackupStatus",
+				Reason:             string(operatorv1alpha1.BackupPending),
+				Message:            fmt.Sprintf("%s", job.Name),
+				Status:             v1.ConditionTrue,
+				LastTransitionTime: metav1.NewTime(time.Now()),
+			},
+			{
+				Type:               "BackupFilename",
+				Reason:             string(operatorv1alpha1.BackupPending),
+				Message:            backupFileName,
+				Status:             v1.ConditionTrue,
+				LastTransitionTime: metav1.NewTime(time.Now()),
```

Comment:
As mentioned above if we go with type==reason for simplicity, then this can just be one condition with `Type: BackupPending` and `Reason: BackupPending` as well.
Otherwise `BackupStatus` type could logically have multiple reasons, e.g pending, failed, completed etc.

And the filename isn't really a condition or phase of the backup right? It's more just info about where we intend to save the backup (not guaranteed to even succeed yet).
WDYT about just including that in the message for the pending condition e.g:
```suggestion
				Type:               operatorv1alpha1.BackupPending,
				Reason:           string(operatorv1alpha1.BackupPending),
				Message:        fmt.Sprintf("Executing job %s to save backup file %s", job.Name, backupFileName),
				Status:             v1.ConditionTrue,
				LastTransitionTime: metav1.NewTime(time.Now()),
			},
```

Reply:
applied. The reason to have the filename as its own message is to have an easier time for retention later